types/react/index.d.ts

```diff
@@ -264,6 +265,25 @@ declare namespace React {
         props?: Q, // should be Q & Attributes
         ...children: ReactNode[]): ReactElement<P>;
 
+    // Context via RenderProps
+    type Provider<T> = ComponentType<{
+        value: T,
+        children?: ReactNode
```

Comment:
Should the children be made mandatory? Not much point in a Provider without _any_ children.

Reply:
Would it really matter since `ReactNode` already accounts for `undefined`? I can see a scenario where someone may not render an app while waiting for an API result but would still like to layer a context provider on top of it (`undefined` or `null` children)

Reply:
ReactNode is wrong altogether, it should not include `undefined`. See this https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23422#pullrequestreview-93993569 from Dan. It should be better eventually when this is resolved ... https://github.com/Microsoft/TypeScript/issues/21699. However, for a preparation for that I believe that `children` should be indeed mandatory.

Reply:
Yeah, `ReactNode` is definitely wrong. If you _actually_ use `ReactNode` anywhere, you'll get a warning from React.

I think the only time it doesn't warn is if you put it as an array element, but otherwise it's a warning.

Reply:
It has to be `ReactNode` due to the above TypeScript issue though, otherwise it's impossible to use React with `noImplicitAny`. After the above issue is resolved this and many more types can be cleaned up but right now there is no option.

Reply:
Side note, I do have a custom alias for this so I can actually have render prop SFC components and I am ignoring the type from ofic. typings. `declare type ReactNode = React.ReactElement<any> | null`. When this is resolved correctly, I can easily update this alias. It's useful here, but works well otherwise :)